### PR TITLE
chore(CLI-integ-tests): cli integ tests cannot use local CDK framework

### PR DESCRIPTION
**Pull request by comcalvi**
### Reason for this change

The cli integration tests cannot use your local version of `aws-cdk-lib`. This can be verified by making your `Stack` construct throw an error upon creation, and watching no CLI integration tests fail, even with `-a`.

### Description of changes

Fixed the CLI integration test framework to correctly link the local packages, like it [used to](https://github.com/aws/aws-cdk/pull/23590).

### Description of how you validated changes

Manual testing. This isn't something we can add automated tests for. 

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/cli-integ/lib/package-sources/repo-source.ts [hunk 1]
```diff
@@ -75,13 +75,14 @@ const YARN_MONOREPO_CACHE: Record<string, any> = {};
...
+export async function findYarnPackages(root: string): Promise<Record<string, string>> {
   if (!(root in YARN_MONOREPO_CACHE)) {
-    const output: YarnWorkspacesOutput = JSON.parse(await shell(['yarn', 'workspaces', '--silent', 'info'], {
+    const outputDataString: string = JSON.parse(await shell(['yarn', 'workspaces', '--json', 'info'], {
       captureStderr: false,
       cwd: root,
       show: 'error',
-    }));
+    })).data;
+    const output: YarnWorkspacesOutput = JSON.parse(outputDataString);
```
Comment (scanlonp) [inline packages/@aws-cdk-testing/cli-integ/lib/package-sources/repo-source.ts:85]: Why do we need to parse the shell output again?

File: packages/@aws-cdk-testing/cli-integ/lib/package-sources/repo-source.ts [hunk 2]
```diff
@@ -96,7 +97,7 @@ async function findYarnPackages(root: string): Promise<Record<string, string>> {
  * Find the root directory of the repo from the current directory
  */
 export async function autoFindRoot() {
-  const found = await findUp('release.json');
+  const found = findUp('release.json');
```
Comment (scanlonp) [inline packages/@aws-cdk-testing/cli-integ/lib/package-sources/repo-source.ts:100]: Odd that this was ever awaited. Looking in the `findUp` function, I am not seeing async calls, but is there any chance this await is necessary?

**Review by scanlonp — COMMENTED** [review]
This essentially skips install of the CDK and uses the local copy, correct? Just getting clarification, then good to approve.

File: packages/@aws-cdk-testing/cli-integ/lib/package-sources/repo-source.ts (hunk 2 above)
Comment (comcalvi) [reply packages/@aws-cdk-testing/cli-integ/lib/package-sources/repo-source.ts:100]: TS compiler said it's not needed, so I removed it. It doesn't return a promise, so there's no need to await anything here.

File: packages/@aws-cdk-testing/cli-integ/lib/package-sources/repo-source.ts (hunk 1 above)
Comment (comcalvi) [reply packages/@aws-cdk-testing/cli-integ/lib/package-sources/repo-source.ts:85]: we first need to parse the output of `yarn workspaces`, which includes a `data` object. That `data` object is an escaped JSON string, which needs to be parsed again.

**Comment by comcalvi** [comment]
Yep, it skips installing the CDK and uses the local version.

**Review by scanlonp — APPROVED** [review]

**Pull request merged**